toplevel/topcommon.ml

```diff
@@ -199,10 +199,11 @@ let preprocess_phrase ppf phr =
   let phr =
     match phr with
     | Ptop_def str ->
-        let str =
-          Pparse.apply_rewriters_str ~restore:true ~tool_name:"ocaml" str
+        let impl =
+          Ast_helper.Impl.mk str
+          |> Pparse.apply_rewriters_impl ~restore:true ~tool_name:"ocaml"
         in
-        Ptop_def str
+        Ptop_def impl.pimpl_structure
```

Comment:
If I understand correctly, this whole change should disappear if `apply_rewriters_str` is kept around